subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt

```diff
@@ -244,6 +258,7 @@ class ResidualProgramCompiler(
             NEW(precompiledBuildscriptWithPluginsBlock)
             ALOAD(Vars.ScriptHost)
             // ${plugins}(temp.createSpec(lineNumber))
+            // TODO: JLL: Passing null here feels like a hack.
             emitPluginRequestCollectorCreateSpecFor(plugins)
```

Comment:
@bamboo and/or @eskatos I'd appreciate your insights here on what I need to change to not need to pass `null` here, or if you have a better solution in general.